homeassistant/components/govee_light_api/light.py

```diff
@@ -0,0 +1,161 @@
+"""Govee Light."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from govee_local_api import GoveeDevice, GoveeLightCapability
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP_KELVIN,
+    ATTR_RGB_COLOR,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import GoveeLocalApiCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Govee light setup."""
+
+    coordinator: GoveeLocalApiCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    def discovery_callback(device: GoveeDevice, is_new: bool) -> bool:
+        if is_new:
+            async_add_entities([GoveeLight(coordinator, device)])
+        return True
+
+    async_add_entities(
+        GoveeLight(coordinator, device) for device in coordinator.devices
+    )
+
+    await coordinator.set_discovery_callback(discovery_callback)
+
+
+class GoveeLight(CoordinatorEntity[GoveeLocalApiCoordinator], LightEntity):
+    """Govee Light."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        coordinator: GoveeLocalApiCoordinator,
+        device: GoveeDevice,
+    ) -> None:
+        """Govee Light constructor."""
+
+        super().__init__(coordinator)
+        self._device = device
+        self._device.set_update_callback(self._update_callback)
+
+        self._attr_unique_id = device.fingerprint
+
+        capabilities = device.capabilities
+        color_modes = set()
+        if capabilities:
+            if GoveeLightCapability.COLOR_RGB in capabilities:
+                color_modes.add(ColorMode.RGB)
+            if GoveeLightCapability.COLOR_KELVIN_TEMPERATURE in capabilities:
+                color_modes.add(ColorMode.COLOR_TEMP)
+                if ColorMode.COLOR_TEMP in color_modes:
```

Comment:
This if statement doesn't do anything? This expression is always made True the line before